tools/walletextension/rpcapi/utils.go

```diff
@@ -207,13 +207,19 @@ func withCache[R any](cache cache.Cache, cfg *CacheCfg, cacheKey []byte, onCache
 		return onCacheMiss()
 	}
 
+	isEvictable := false
 	ttl := longCacheTTL
 	if cacheType == LatestBatch {
 		ttl = shortCacheTTL
+		isEvictable = true
 	}
 
 	cachedValue, foundInCache := cache.Get(cacheKey)
-	if foundInCache && !cache.IsEvicted(cacheKey, ttl) {
+
+	// only entries cached with `LatestBatch` are evicted
+	isEvicted := isEvictable && cache.IsEvicted(cacheKey, ttl)
```

Comment:
I find this confusing

Reply:
I'll  clarify the comment. Hopefully that works.